xmtp/src/client.rs

```diff
@@ -34,71 +42,111 @@ where
     pub persistence: NamespacedPersistence<P>,
     pub(crate) account: Account,
     pub(super) _store: S,
-}
 
-impl<A, P, S> Client<A, P, S>
-where
-    A: XmtpApiClient,
-    P: Persistence,
-{
+    is_initialized: bool,
 }
 
 impl<A, P, S> Client<A, P, S>
 where
     A: XmtpApiClient,
     P: Persistence,
 {
-    pub fn write_to_persistence(&mut self, s: &str, b: &[u8]) -> Result<(), P::Error> {
-        self.persistence.write(s, b)
-    }
-
-    pub fn read_from_persistence(&self, s: &str) -> Result<Option<Vec<u8>>, P::Error> {
-        self.persistence.read(s)
+    pub fn new(
+        api_client: A,
+        network: Network,
+        persistence: NamespacedPersistence<P>,
+        account: Account,
+        store: S,
+    ) -> Self {
+        Self {
+            api_client,
+            network,
+            persistence,
+            account,
+            _store: store,
+            is_initialized: false,
+        }
     }
 
     pub fn wallet_address(&self) -> Address {
         self.account.addr()
     }
 
-    pub async fn get_contacts(
+    async fn noop(&self) {}
+    pub async fn init(&mut self) -> Result<(), ClientError> {
+        // Register Contact Bundles
+        let registered_bundles = self.noop().await; // Causes #[swift_bridge::bridge] to fail when noop is a member fn
+
```

Comment:
Calling an async member function in `init` causes errors in the swift bridge. It appears that the self parameter included in the async call adds a reference of Client to the future. 

This results in client being required to implement Sync. 

The current test configuration uses an ephemeral Diesel implementation which does not implement Sync (because it is not threadsafe).

Reply:
https://github.com/xmtp/libxmtp/actions/runs/5072644086/jobs/9110613123?pr=121